resources/css/bem/mod.less

```diff
@@ -2,89 +2,184 @@
+      // `.Darken(2.8f)` amounts to dividing every colour component by 1 + 2.8 = 3.8
+      // which is the same thing as interpolating between black and @colour at t = 1 / 3.8 ≈ 0.2631578947
+      // using `srgb` is intentional because the `.Darken()` helper does no colourspace conversion
+      --type-extender-colour: color-mix(in srgb, black, @colour 26.3%);
     }
   }
 
-  .all(~"4K", "4K");
-  .all(~"5K", "5K");
-  .all(~"6K", "6K");
-  .all(~"7K", "7K");
-  .all(~"8K", "8K");
-  .all(~"9K", "9K");
-  .all(~"AP", "autopilot");
-  .all(~"CL", "classic");
-  .all(~"DT", "double-time");
-  .all(~"EZ", "easy");
-  .all(~"FI", "fader");
-  .all(~"FL", "flashlight");
-  .all(~"HD", "hidden");
-  .all(~"HR", "hard-rock");
-  .all(~"HT", "half");
-  .all(~"MR", "mirror");
-  .all(~"NC", "nightcore");
-  .all(~"NF", "no-fail");
-  .all(~"NM", "no-mod");
-  .all(~"PF", "perfect");
-  .all(~"RD", "random");
-  .all(~"RX", "relax");
-  .all(~"SD", "sudden-death");
-  .all(~"SO", "spun-out");
-  .all(~"TD", "touchdevice");
-  .all(~"V2", "v2");
-
-  .type-bg(Automation);
-  .type-bg(Conversion);
-  .type-bg(DifficultyIncrease);
-  .type-bg(DifficultyReduction);
-  .type-bg(Fun);
-
-  &::before {
-    .full-size();
-    background-image: var(--type-bg);
+  .mod-type(Automation, @osu-colour-blue-1);
+  .mod-type(Conversion, @osu-colour-purple-1);
+  .mod-type(DifficultyIncrease, @osu-colour-red-1);
+  .mod-type(DifficultyReduction, @osu-colour-lime-1);
+  .mod-type(Fun, @osu-colour-pink-1);
+  .mod-type(System, @yellow);
+
+  &__icon {
+    position: relative;
+    width: 1.42em;
     background-size: contain;
     background-position: center;
     background-repeat: no-repeat;
-    content: "";
-    display: var(--generic-display);
 
-    @media @highdpi {
-      background-image: var(--type-bg-2x);
+    &::before {
+      .full-size();
+      background-color: var(--type-bg-colour);
+      mask-image: url("~@images/badges/mods/blanks/mod-icon.svg");
+      mask-size: contain;
+      mask-position: center;
+      mask-repeat: no-repeat;
+      content: "";
+      display: flex;
+    }
+
+    &::after {
+      .full-size();
+      content: attr(data-acronym);
+      display: flex;
+      align-items: center;
+      justify-content: center;
+      color: var(--type-fg-colour);
+      font-family: @font-grade;
+      font-weight: 900;
+      font-size: 0.4em;
+    }
+
+    .mod-icon-osu(@acronym, @filename) {
+      &--@{acronym}::after {
+        content: "";
+        background-color: var(--type-fg-colour);
+        mask-image: url("~@images/badges/mods/@{filename}.svg");
+        mask-position: center;
+        mask-size: contain;
+        mask-repeat: no-repeat;
+        margin: 0;
+      }
+    }
+
+    .mod-icon-fa(@acronym, @glyph) {
+      display: flex;
+      content: @glyph;
+      font-size: 0.8em;
+      // slight adjustments for small sizes
+      padding-left: 1px;
+      padding-bottom: 1px;
```

Comment:
I don't think it works for small sizes...

![](https://s.kohaku.love/2025/07/chrome_2025-07-11_20-28-39.png)

compared to no padding-bottom:

![](https://s.kohaku.love/2025/07/chrome_2025-07-11_20-29-16.png)